modules/apps/collaboration/invitation/invitation-invite-members-web/src/main/java/com/liferay/invitation/invite/members/web/internal/portlet/InviteMembersPortlet.java

```diff
@@ -232,6 +236,18 @@ protected void doSendInvite(ActionRequest actionRequest) throws Exception {
 		String createAccountURL = _portal.getCreateAccountURL(
 			request, themeDisplay);
 
+		Layout layout = themeDisplay.getLayout();
+
+		if (layout.isPrivateLayout()) {
+			StringBundler sb = new StringBundler(3);
+
+			sb.append(_portal.getPortalURL(themeDisplay));
+			sb.append(StringPool.QUESTION);
+			sb.append(HttpUtil.getQueryString(createAccountURL));
+
+			createAccountURL = sb.toString();
+		}
+
```

Comment:
Hey @IstvanD 

The failure seems to be in the _portal.getCreateAccountURL, not in the view. I can't see why a method called like this do not send us to the  create account page in some cases??

Saying this, could the logic be moved to this method implementation??

Thanks!

Reply:
Hey @robertoDiaz 

Examining `PortalImpl.getCreateAccountURL()` I see that it is possible to add a custom layout for strangers to create accounts.

I have set the private page's friendly URL as

company.security.strangers.url=/privatepage

The problem is, that
`
			Layout layout = LayoutLocalServiceUtil.getFriendlyURLLayout(
				themeDisplay.getScopeGroupId(), false,
				PropsValues.COMPANY_SECURITY_STRANGERS_URL);
`

works only for public layouts (privateLayout = false), so when the invitation is sent from a private page, this will lead to a `NoSuchLayoutException`.

Reply:
I do not fully understand your last comment.

I was talking about copy and paste the jsp logic in the method.